app/models/keyword.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class Keyword < ApplicationRecord
+  enum status: { processing: 'processing', succeeded: 'succeeded', failed: 'failed' }
+
+  has_many :keyword_search_entries, dependent: :destroy
+
+  validates :content, :status, presence: true
+  validates :result_page_html, presence: true, if: :succeeded?
+  validates :links_count,
+            presence: true,
+            numericality: { greater_than_or_equal_to: 0 },
+            if: :succeeded?
+
+  private
+
+  def succeeded?
```

Comment:
with [Enum](https://api.rubyonrails.org/v7.0.6/classes/ActiveRecord/Enum.html), it will define the `succeeded?` method for us  FYI, so we can remove this one ;)

Reply:
Fixed in 3f79f1904b1d6154608ebca568f0920f040814ec